### DISCUSSION TITLE
Does osgEarth work with osg in GL1/GL2 mode?

### DISCUSSION THREAD

**Discussion by IngoKleefeld** [pelicanmapping/osgearth#2257] [Support Q&A] [answered]
I compile osg with the default settings of cmake. OpenGL default is GL1/GL2 on, GL3 off. When I start my little test program, it always crashes somewhere in the depths of osgEarth rex_engine->xtree. The same happens when I start osgearth_city.exe, crash also rex_engine->xtree. But what is amazing is that osgearth_viewer.exe works. Does osgEarth basically not work in GL1/GL2 mode anymore or is this still a bug in osgEarth? Or what am I doing wrong?

I use osgEarth 3.3 with osg 3.6.5.

My simple test app:
		osgEarth::initialize();

		// create the map.
		osg::ref_ptr<Map> map = new Map();

		// add a TMS imagery layer:
		TMSImageLayer* layer = new TMSImageLayer();
		layer->setURL("...../EarthMaps/BlueMarble/tms.xml");
		layer->setTMSType("google");
		map->addLayer(layer);

		// initialize a viewer:
		osgViewer::Viewer viewer;

		VDGEarthManipulator* manip = new VDGEarthManipulator();
		viewer.setCameraManipulator( manip );

		osg::Group* root = new osg::Group();
		viewer.setSceneData( root );

		// make the map scene graph:
		MapNode* mapNode = new MapNode(map.get());
		root->addChild( mapNode );

		return viewer.run();

**Comment by gwaldron** [comment, marked as answer]
"Officially" the SDK requires OSG with GL3.
But as you saw, some basic things will still work.
